plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/IntegerProjection.java

```diff
@@ -0,0 +1,90 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.aws.athena.projection;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.IntegerType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static java.util.Objects.requireNonNull;
+
+public class IntegerProjection
+        extends Projection
+{
+    private final int leftBound;
+    private final int rightBound;
+    private final int interval;
+    private final Optional<Integer> digits;
+
+    public IntegerProjection(String columnName, int leftBound, int rightBound, int interval, Optional<Integer> digits)
+    {
+        super(columnName);
+        this.leftBound = leftBound;
+        this.rightBound = rightBound;
+        this.interval = interval;
+        this.digits = requireNonNull(digits, "digits is null");
+    }
+
+    @Override
+    public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
+    {
+        Builder<String> builder = ImmutableList.builder();
+        int current = leftBound;
+        while (current <= rightBound) {
+            String currentValue = Integer.toString(current);
+            String alignedValue = digits
+                    .map(digits -> alignDigitPlace(digits, currentValue))
+                    .orElse(currentValue);
+            if (isValueInDomain(partitionValueFilter, current, alignedValue)) {
+                builder.add(alignedValue);
+            }
+            current += interval;
+        }
+        return builder.build();
+    }
+
+    private boolean isValueInDomain(Optional<Domain> valueDomain, int value, String formattedValue)
+    {
+        if (valueDomain.isEmpty() || valueDomain.get().isAll()) {
+            return true;
+        }
+        Domain domain = valueDomain.get();
+        Type type = domain.getType();
+        if (type instanceof VarcharType) {
+            return domain.contains(singleValue(type, utf8Slice(formattedValue)));
+        }
+        if (type instanceof IntegerType || type instanceof BigintType) {
+            return domain.contains(singleValue(type, Long.valueOf(value)));
+        }
+        return false;
+    }
+
+    private String alignDigitPlace(int digits, String value)
```

Comment:
this looks like `String.format("%0" + digits + "d", value)`

Reply:
Yes but we might get situations where digits is negative, so I prefere to make it straight forward.

Reply:
I saw it as a little less clear to me. But it reduces code from one method so Appliec

Reply:
What would negative digits mean? Should we handle it somehow. If not we should throw exception if we get it, right?

Reply:
Code is no longer here as we use your solution. But negative value would just indicate that value has more characters than we align to. And yes that's potentially a bad situation. 
Nevertheless now we use int using String.format.